lib/node_modules/@stdlib/repl/test/integration/test.syntax_highlighting.js

```diff
@@ -0,0 +1,203 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var DebugStream = require( '@stdlib/streams/node/debug' );
+var replace = require( '@stdlib/string/replace' );
+var readDir = require( '@stdlib/fs/read-dir' ).sync;
+var format = require( '@stdlib/string/format' );
+var repl = require( './fixtures/repl.js' );
+
+
+// VARIABLES //
+
+var RE_JSON = /\.json$/;
+
+var FIXTURES_DIR = resolve( __dirname, 'fixtures', 'syntax-highlighting' );
+var FIXTURES_FILES = filter( readDir( FIXTURES_DIR ) );
+
+
+// FUNCTIONS //
+
+/**
+* Filters a list of files for those having a `*.json` file extension.
+*
+* @private
+* @param {Array<string>} list - file list
+* @returns {Array<string>} filtered list
+*/
+function filter( list ) {
+	var out;
+	var i;
+
+	out = [];
+	for ( i = 0; i < list.length; i++ ) {
+		if ( RE_JSON.test( list[ i ] ) ) {
+			out.push( list[ i ] );
+		}
+	}
+	return out;
+}
+
+/**
+* Returns default settings.
+*
+* @private
+* @returns {Object} default settings
+*/
+function defaultSettings() {
```

Comment:
Note for future selves, we may want to figure out a way to abstract this out of the various test files. Currently, each time we add a setting, we're having to update all test files accordingly which may not be sustainable in the long term.